test/ref_ops_test.cpp

```diff
@@ -4992,6 +4992,27 @@ TEST_CASE(reduce_max_axis0)
     EXPECT(results_vector == gold);
 }
 
+TEST_CASE(reduce_max_dynamic_axis0)
+{
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+    migraphx::shape s{migraphx::shape::float_type, {{2, 4, 2}, {3, 5, 3}}};
+    auto input = mm->add_parameter("X", s);
+    std::vector<float> input_data{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12};
```

Comment:
just as a convention, since this is not used to build and compile the program, move this declaration of `input_data` where you are setting `params` later.